### PR TITLE
Improve Circle and Avatar to use specific size

### PR DESCRIPTION
**Pull request by JF-Cozy**
We would like to add a 24px size for the Avatars => we can now use specific size for avatars.
There was also a need to correct some overlapping behavior.

Circle is improved as well

Avatar : https://jf-cozy.github.io/cozy-ui/react/#!/Avatar
Circle : https://jf-cozy.github.io/cozy-ui/react/#!/Circle

### CODE REVIEW

**Comment by ptbrowne** [comment]
I think we should have a component for the Avatar "chain" with overlap. Also, in the example, the avatars that overlap seem to lack a white border and/or box-shadow something to add a depth feelink ? @joel-costa

**Comment by joel-costa** [comment]
> I think we should have a component for the Avatar "chain" with overlap. Also, in the example, the avatars that overlap seem to lack a white border and/or box-shadow something to add a depth feelink ? @joel-costa

Yep 👍
I don't know if it is something to provide in the component but I think it is interesting to note that ideally, white borders would not stay white when we switch the background-color of his container : 
image

**Comment by JF-Cozy** [comment]
I will add the Avatar "chain" in the overlap exemple.

Overlap is a "forced" example just to show that it's not buggy if you do it (which you do in cozy-sharing). It's not something that can be used turnkey.

File: react/Avatar/index.jsx [hunk 1]
```diff
@@ -60,23 +61,34 @@ export const Avatar = ({
...
-  size: PropTypes.oneOf(['xsmall', 'small', 'medium', 'large', 'xlarge']),
-  className: PropTypes.string
+  size: PropTypes.oneOf([
+    'xsmall',
+    'smaller',
+    'small',
+    'medium',
+    'large',
+    'xlarge'
+  ]),
```
Comment (Crash--) [inline react/Avatar/index.jsx:79]: Hard to follow all this size. Why not starting to add "size=24" instead of "smaller" ?

File: stylus/components/avatar.styl [hunk 2]
```diff
@@ -3,6 +3,8 @@
 $avatar
     background-color var(--paleGrey)
     color var(--silver)
+    position relative
+    z-index 1
```
Comment (Crash--) [inline stylus/components/avatar.styl:7]: we do have a z-index.styl with css var. We should use for instance `$low-index` instead

**Review by Crash-- — COMMENTED** [review]
Wondering if we should not provide in UI a component to display an AvatarsList 🤔

File: react/Avatar/index.jsx (hunk 1 above)
Comment (ptbrowne) [reply react/Avatar/index.jsx:79]: Yes. @joel-costa  do we really need 6 sizes of Avatar ? I think we should have less sizes and the ability for the user to give a custom size.

Comment (joel-costa) [reply react/Avatar/index.jsx:79]: @ptbrowne : Size doesn't matter for you ? :p
You would have only 3 sizes like on MUI ? No worries for me if we can give a custom size. Even on the home, we have a specific size : 
image

But I don't know if it would be relevant to set the same 3 sizes as MUI (24 - 40 - 56). 
If we exclude the home, we only use 16 - 24 - 32 - 40 right ?

**Comment by JF-Cozy** [comment]
For now, I will focus on this point: to be able to give a specific size in pixels

File: react/Circle/index.jsx [hunk 3]
```diff
@@ -1,28 +1,34 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 import cx from 'classnames'
+
 import styles from './styles.styl'
 
+export const createSizeStyle = size => {
+  const sizeToPixel = Number.isInteger(size)
+    ? size
+    : { xsmall: 16, small: 32, medium: 40, large: 48, xlarge: 64 }[size]
```
Comment (ptbrowne) [inline react/Circle/index.jsx:10]: Nit: The mapping can be put outside the function

**Review by ptbrowne — APPROVED** [review]
